### PR TITLE
init: jni exception handling

### LINKED ISSUE #65: [GraphSDK] implement exception and error handling between java and jni
Setup the exception handling between Rust -> JNI -> Java

Should allow throwing exceptions from inside rust jni code.

Related to #28

### PR DESCRIPTION
**Pull request by aramikm**
# Goal
The goal of this PR is to implement exception handing for jni

Closes #65 

# Discussion
- created Java specific exceptions
- throwing exceptions from jni side
- changed mutex to RWLock for memory states
- some refactor and clean up in apis
- Added input validation in api

# Checklist
- [x] added tests

### CODE REVIEW

File: bridge/jni/src/errors.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
+use dsnp_graph_config::errors::DsnpGraphError;
+use jni::{objects::JThrowable, JNIEnv};
+use std::fmt;
+
+#[derive(Debug)]
+pub enum SdkJniError {
+	DsnpGraph(DsnpGraphError),
```
Comment (saraswatpuneet) [inline bridge/jni/src/errors.rs:7]: Not sure if in JNI we should also return the `error_codes` we have set

Comment (aramikm) [reply bridge/jni/src/errors.rs:7]: Great idea

File: bridge/jni/src/errors.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+impl From<protobuf::Error> for SdkJniError {
+	fn from(e: protobuf::Error) -> SdkJniError {
+		SdkJniError::InvalidProto(e)
+	}
+}
+
+impl fmt::Display for SdkJniError {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		match self {
+			SdkJniError::DsnpGraph(s) => write!(f, "ErrorCode({}) {}", s.error_code(), s),
```
Comment (aramikm) [inline bridge/jni/src/errors.rs:57]: error code exposes in message

File: .github/workflows/pr-verify.yml [hunk 3]
```diff
@@ -112,39 +112,39 @@ jobs:
...
-        run: make test-ffi
-      - name: Upload artifacts dsnp_graph_sdk_ffi.h and libdsnp_graph_sdk_ffi.a
-        uses: actions/upload-artifact@v2
-        with:
-          name: dsnp_graph_sdk_ffi
-          path: |
-            bridge/ffi/src/c_example/dsnp_graph_sdk_ffi.h
-            bridge/ffi/src/c_example/libdsnp_graph_sdk_ffi.a
-            bridge/ffi/src/c_example/main.c
+#  test_binary:
```
Comment (aramikm) [inline .github/workflows/pr-verify.yml:115]: temporarily disabling ffi-tests job since they are fixed

**Pull request merged**